bokeh/io.py

```diff
@@ -57,10 +57,29 @@
 #-----------------------------------------------------------------------------
 
 class _CommsHandle(object):
+
+    _json = {}
+
     def __init__(self, comms, doc, json):
+        self._cellno = None
+        try:
+            from IPython import get_ipython
+            ip = get_ipython()
+            hm = ip.history_manager
+            p_prompt = list(hm.get_tail(1, include_latest=True))[0][1]
+            self._cellno = p_prompt
+        except:
```

Comment:
I did think about it but was not sure what that would buy in this instance. In this case I am just trying to keep it from falling over completely if Jupyter changes something out from under us.